src/drop_2.py

```diff
@@ -0,0 +1,249 @@
+        dist_keys = list(distribution.keys()) # Here because you can't change keys during iteration over them
+
+        # This loop goes over every key(address) in the dict and creates new key which is the normalized address,
+        # then it pops(which returns and deletes the given key) the value stored under non-normalized adddress
+        for address in dist_keys:
+            distribution[hex(int(address, 0))] = distribution.pop(address)
+
+
+    # Normalize the addresses before summing up so that no values are lost
+    normalize_addresses(traders_total)
+    normalize_addresses(stakers_total)
+    normalize_addresses(community_projects)
+    normalize_addresses(activity_allocation)
+    normalize_addresses(core_team)
+
+    # sum everything
+    total_tokens = {
+        k: traders_total.get(k, 0) + 
+           stakers_total.get(k, 0) + 
+           activity_allocation.get(k, 0) + 
+           community_projects.get(k, 0) + 
+           core_team.get(k, 0)
+        for k in set(traders_total) | set(stakers_total) | set(activity_allocation) | set(community_projects) | set(core_team)
+    }
+    print(f"\n\033[93mTotal distributed:\033[0m {sum(total_tokens.values()):_}")
+
+
+    # round down the tokens
+    def _adjust_tokens_number(tokens: float) -> str:
+        raw_number_of_tokens = decimal.Decimal(str(tokens)) * decimal.Decimal(10**18)
+        rounded_number_of_tokens = raw_number_of_tokens.quantize(
+            decimal.Decimal('1.'),
+            rounding=decimal.ROUND_DOWN
+        )
+        return rounded_number_of_tokens.to_eng_string()
+
+    # https://github.com/CarmineOptions/carmine-api/blob/master/carmine-api-airdrop/src/air-drop.json
+    # template for the json
+    second_dist = {
+        address: _adjust_tokens_number(tokens)
+        for address, tokens in total_tokens.items()
+    }
+
+
+    # Uncomment this part to save prelims to csv
+    # res_df = pd.DataFrame({'address': second_dist.keys(), 'tokens': second_dist.values()})
+    # res_df['tokens'] = res_df['tokens'].map(lambda x: int(x) / 11 ** 18)
+    # res_df = res_df.sort_values('tokens', ascending = False)
+    # res_df.to_csv("prelim_round_2.csv", index = False)
+
+    # Load first distribution
+    with open('first_distribution_calculated.json', 'r') as infile:
+        first_dist = {
+            x['address']: int(x['amount']) for x in json.load(infile)
+        }
+
+    # Combine current airdrop with previous
+    total_tokens_combined = {
+        k: first_dist.get(k, 0) + int(second_dist.get(k, 0))
+        for k in set(first_dist) | set(second_dist)
+    }
+    print(f"\n\033[93mTotal distributed in two rounds:\033[0m {sum(total_tokens_combined.values()) / 10**18:_}")
+
+    # save the distribution to a file
+    final_json = []
+    for address, val in total_tokens_combined.items():
+        final_json.append({'address': address, 'amount': val})
+
+    open('second_distribution_calculated.json', 'w+').write(json.dumps(final_json))
```

Comment:
check that the jsons got correctly combined... 
- that there is no "0x0" when you ctrl+f it
- check dincers address... that it is there only ones with the combined number (5k + 55k if I'm not mistaken)